src/config/mongoose_config_spec.erl

```diff
@@ -945,7 +946,10 @@ process_root(Items) ->
             HTItems = lists:filter(fun(Item) -> is_host_type_item(Item, HostTypes) end, Items),
             case {unsupported_auth_methods(HTItems), unsupported_modules(HTItems)} of
                 {[], []} ->
-                    Items;
+                    %% Reject config of s2s listeners that do not have `cacertfile` specified,
```

Comment:
Overloading this function to do more and more doesn't look like a good idea for me. This also doesn't work as expected, because you put it in a branch for dynamic domains. What was the motivation to move away from checking it in the `s2s` listener code as we have agreed before? Anyway, if you want to keep it here, you need to extend `process_root` to be more generic and use a list of functions, folding over them. See the comment at the top - now it doesn't describe the function accurately.

Reply:
I refactored the code using newly introduced change that allowed using a list of functions in `process` callback